src/gui/buttons.py

```diff
@@ -42,17 +83,32 @@ def _handle(event: UIEvent):
     return _handle
 
 
-def nav_button(target: type[arcade.View], text: str) -> UIFlatButton:
+def nav_button(target: type[arcade.View], text: str) -> UITextureButton:
     """A generic button for changing to a different View.
 
     Args:
         target (type[arcade.View]): The target View to change to, passed to the click handler of the button.
         text (str): The text to render on the button, typically the name of the view that will be displayed on click.
 
     Returns:
-        UIFlatButton: A button with a text label and an attached click handler.
+        UITextureButton: A button with a text label and an attached click handler.
     """
-    btn = UIFlatButton(text=text)
+    main_tex = PixelatedNinePatch(
+        left=1, right=1, bottom=3, top=1, texture=WindowData.buttons[7]
+    )
+    pressed_tex = PixelatedNinePatch(
+        left=1, right=1, bottom=3, top=3, texture=WindowData.buttons[9]
+    )
+    hovered_tex = PixelatedNinePatch(
+        left=1, right=1, bottom=3, top=1, texture=WindowData.buttons[11]
+    )

```

Comment:
These should probably be supplied as arguments since they are implementation specific

Reply:
Or use-case specific rather

Reply:
Actually now that I've looked at more of the code, this should be defined as a dict at the module level, see example on the `end_turn_button` function